### PR TITLE
Merge ubuntu and macos fuzzer && Record time for `./test_all_fuzzers.sh`

### PR DESCRIPTION
**Pull request by syheliel**
1. merge ubuntu fuzzers and macos fuzzers
2. add ccache(no obvious time boost)
3. record time for each fuzzer when running `./test_all_fuzzers.sh`

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -94,23 +94,38 @@ jobs:
...
     steps:
     - uses: actions-rs/toolchain@v1
       with:
         profile: minimal
         toolchain: stable
+    - name: add ccache 
+      uses: hendrikmuhs/ccache-action@v1.2
+      with:
+        key: ${{ github.job }}-${{ matrix.os }}
+        max-size: 2000M
```
Comment (domenukk) [inline .github/workflows/build_and_test.yml:113]: Maybe we want to increase the size?

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -182,25 +197,25 @@ jobs:
...
-    - name: Install deps
-      run: brew install llvm libpng nasm coreutils z3 && brew link --force llvm
-    - name: install cargo-make
-      run: cargo install --force cargo-make
-    - uses: actions/checkout@v2
-    - name: Increase map sizes
-      run: ./scripts/shmem_limits_macos.sh
-    - name: Build and run example fuzzers
-      run: ./scripts/test_all_fuzzers.sh
+  # macos-fuzzers:
```
[the remark this answers is not in the document]
Comment (syheliel) [reply .github/workflows/build_and_test.yml]: OK

File: .github/workflows/build_and_test.yml (hunk 1 above)
Comment (syheliel) [reply .github/workflows/build_and_test.yml:113]: Previous run shows that it only takes about ~200M, so `max-size: 2000M` is enough.
The lastest run add ccache to `CXX`, wait for the result to see if we should adjust the size.

File: scripts/test_all_fuzzers.sh [hunk 3]
```diff
@@ -12,9 +12,12 @@ libafl=$(pwd)
 
 git submodule init && git submodule update
 
+# record time of each fuzzer
+declare -A time_record || (echo "declare -A not avaliable, please update your bash bersion to 4";exit 1)
```
Comment (tokatoka) [inline scripts/test_all_fuzzers.sh:16]: typo here

Comment (syheliel) [reply scripts/test_all_fuzzers.sh]: Thanks, Fixed.

**Comment by domenukk** [comment]
Can we somehow include the ccache directory (`/home/runner/work/LibAFL/LibAFL/.ccache`) to persist between runs, or is that already the case?
I see qemu fuzzers take a very long time to build. Maybe we want to specifically cache the submodule between runs, too?

**Comment by syheliel** [comment]
@domenukk 
cache for `ccache` has been set when using `hendrikmuhs/ccache-action@v1.2`.

Have a loook at the log, and here are the most time consuming fuzzers:
```
libfuzzer_stb_image_concolic time:1430
fuzzbench_fork_qemu time:599
libfuzzer_stb_image_sugar time:592
frida_libpng time:506
```
take a quick look at `libfuzzer_stb_image_concolic`, the main bottleneck is building the binary. It's strange that `compiling clap` takes 20mins... 
```
2022-05-14T03:52:30.4284623Z [0m[0m[1m[32m   Compiling[0m clap v3.1.18
2022-05-14T04:08:01.0852346Z [0m[0m[1m[32m   Compiling[0m z3 v0.11.2
```
[1_fuzzers (ubuntu-latest).txt](https://github.com/AFLplusplus/LibAFL/files/8692360/1_fuzzers.ubuntu-latest.txt)

**Pull request merged**